io_scene_godot/converters/armature.py

```diff
@@ -12,9 +12,13 @@ def export_bone_attachment(escn_file, node, parent_gd_node):
     # in the parent armature node
     bone_attachment['bone_name'] = "\"{}\"".format(node.parent_bone)
 
-    # regard to ```export_armature_node()```, the exported bone id
-    # is the index of the bone in node.parent.pose.bones list
-    bone_id = node.parent.pose.bones.find(node.parent_bone)
+    # parent_gd_node is the SkeletonNode with the parent bone
+    bone_id = parent_gd_node.find_bone_id(node.parent_bone)
+    if bone_id == -1:
+        raise ValidationError(
+            "Object '{}' is parented to an unexported bone '{}'"
```

Comment:
Perhaps non-deform bones could check for attached objects before deciding not to be exported?

Reply:
hmm, that sounds better!

Reply:
After looking at it again,  I think it is fine to raise an error like that.  
1. Perform a check on attached objects is not easy,  there are no entries in bone pointing to attached objects
2. object attached to control bone seems useless, raise error is more clear to let user know what they are doing

Reply:
1) Odd that there's no API for it. I can't help feeling that there must be something under-the-hood. Otherwise updating the scene would be awkward. But how about something like:
```for obj in armature.children_recursive:
        if child.bone_parent == bone_of_interest:
            # needed even if non_deform
```

2) On one hand, I agree that it's a case that the user should not do intentionally. But I also think that raising errors is a bit of a pain for the user. It's like the difference between python and C.....

Reply:
maybe iterate through `armature_children` would be enough?  being a child of a bone must be a child of the armature.  
the problem is doing that iteration would harm the code,  do it for every bone waste performance and also need to passing the armature around,  or do it for once before `export_bone`, then all the export bone logic may need to rewrite